common/date_format.py

```diff
@@ -99,36 +105,26 @@ def classify(self, phish):
         sender = self.extract_from(phish)
         date = phish["Date"]
 
-        if sender in self.sender_to_date_data:
-            test_format = DateFormat(date)
-            return self.detect(self.sender_to_date_data[sender], test_format)
+        if not date or not (sender in self.sender_profile):
+            return [0, 0, 0]
 
-        return [0, 0, 0]
+        profile = self.sender_profile[sender]
+        t = date_to_template(date)
 
-    def detect(self, date_data, test_format):
-        curr_zero = list(test_format.zeros_seen)[0]
-        for seen_format in date_data.seen_formats:
-            if (seen_format.same_hash(test_format) and ((curr_zero == 0 or curr_zero in seen_format.zeros_seen) or (len(seen_format.zeros_seen) == 1 and list(seen_format.zeros_seen)[0] == 0))):
-                return [0, date_data.seen_formats[seen_format], sum(date_data.seen_formats.itervalues())]
-        return [1, 0, sum(date_data.seen_formats.itervalues())]
+        if not (t in profile.templates):
+            return [1, 0, profile.num_emails]
+
+        rv = [0, profile.templates[t].total, profile.num_emails]
```

Comment:
For future reference, could you add a comment to the beginning of this function describing what each of these 3 values represent? We haven't been doing this for the other detectors, but I think this will make it easier for people to understand what's happening in each of the detectors as we continue to add new heuristics.